diff --git a/dlls/winemac.drv/display.c b/dlls/winemac.drv/display.c
index da0d1515..c37b8d03 100644
--- a/dlls/winemac.drv/display.c
+++ b/dlls/winemac.drv/display.c
@@ -1482,6 +1482,7 @@ void macdrv_displays_changed(const macdrv_event *event)
 
 static BOOL force_display_devices_refresh;
 
+long forced_updates = 0;
 void macdrv_UpdateDisplayDevices( const struct gdi_device_manager *device_manager,
                                   BOOL force, void *param )
 {
@@ -1491,7 +1492,15 @@ void macdrv_UpdateDisplayDevices( const struct gdi_device_manager *device_manage
     INT gpu_count, adapter_count, monitor_count;
     DWORD len;
 
-    if (!force && !force_display_devices_refresh) return;
+
+    if (!force && !force_display_devices_refresh) {
+        if (forced_updates > 10) {
+            return;
+        } else {
+            forced_updates++;
+        }
+    }
+
     force_display_devices_refresh = FALSE;
 
     /* Initialize GPUs */
