diff --git a/dlls/localspl/provider.c b/dlls/localspl/provider.c
index ca8dc5fb..ce199586 100644
--- a/dlls/localspl/provider.c
+++ b/dlls/localspl/provider.c
@@ -916,6 +916,8 @@ static const  printenv_t * validate_envW(LPCWSTR env)
     unsigned int i;
 
     TRACE("(%s)\n", debugstr_w(env));
+    return all_printenv[1]; // x86_64
+
     if (env && env[0])
     {
         for (i = 0; i < ARRAY_SIZE(all_printenv); i++)
