diff --git a/libexslt/Makefile.am b/libexslt/Makefile.am
index 54495243..7cf5a8c9 100644
--- a/libexslt/Makefile.am
+++ b/libexslt/Makefile.am
@@ -27,7 +27,7 @@ libexslt_la_SOURCES =                   \
        libexslt.h                      \
        dynamic.c

-libexslt_la_LIBADD = $(top_builddir)/libxslt/libxslt.la $(EXTRA_LIBS) $(LIBGCRYPT_LIBS) $(M_LIBS)
+libexslt_la_LIBADD = $(top_builddir)/libxslt/libxslt.la $(EXTRA_LIBS) $(LIBGCRYPT_LIBS) $(M_LIBS) $(LIBXML_LIBS)
 libexslt_la_LDFLAGS = $(WIN32_EXTRA_LDFLAGS) -version-info $(LIBEXSLT_VERSION_INFO)

 man_MANS = libexslt.3