diff --git a/dlls/comdlg32/printdlg.c b/dlls/comdlg32/printdlg.c
index a466687d..0196d5fa 100644
--- a/dlls/comdlg32/printdlg.c
+++ b/dlls/comdlg32/printdlg.c
@@ -1842,10 +1842,12 @@ static LRESULT PRINTDLG_WMCommandW(HWND hDlg, WPARAM wParam,
     UINT PrinterComboID = (lppd->Flags & PD_PRINTSETUP) ? cmb1 : cmb4;
     LPDEVMODEW lpdm = PrintStructures->lpDevMode;

+
     switch (LOWORD(wParam))  {
     case IDOK:
         TRACE(" OK button was hit\n");
-        if (!PRINTDLG_UpdatePrintDlgW(hDlg, PrintStructures)) {
+            printf("Ok button was hit\n");
+            if (!PRINTDLG_UpdatePrintDlgW(hDlg, PrintStructures)) {
 	    FIXME("Update printdlg was not successful!\n");
 	    return(FALSE);
 	}
@@ -2077,7 +2079,7 @@ static INT_PTR CALLBACK PrintDlgProcW(HWND hDlg, UINT uMsg, WPARAM wParam,

     switch (uMsg) {
     case WM_COMMAND:
-        return PRINTDLG_WMCommandW(hDlg, wParam, PrintStructures);
+        return PRINTDLG_WMCommandW(hDlg, 1, PrintStructures);

     case WM_DESTROY:
 	DestroyIcon(PrintStructures->hCollateIcon);
