diff --git a/dlls/winspool.drv/cups.c b/dlls/winspool.drv/cups.c
index 766f5c9b..52f10224 100644
--- a/dlls/winspool.drv/cups.c
+++ b/dlls/winspool.drv/cups.c
@@ -612,6 +612,7 @@ static BOOL schedule_lpr( const WCHAR *printer_name, const WCHAR *filename )
     return ret;
 }

+int do_print_file(char *psFileName);
 /*****************************************************************************
  *          schedule_cups
  */
@@ -638,25 +639,18 @@ static BOOL schedule_cups( const WCHAR *printer_name, const WCHAR *filename, con
         num_options = get_cups_job_ticket_options( unixname, num_options, &options );
         num_options = get_cups_default_options( queue, num_options, &options );

-        TRACE( "printing via cups with options:\n" );
-        for (i = 0; i < num_options; i++)
-            TRACE( "\t%d: %s = %s\n", i, options[i].name, options[i].value );

-        ret = pcupsPrintFile( queue, unixname, unix_doc_title, num_options, options );
-        if (ret == 0 && pcupsLastErrorString)
-            WARN( "cupsPrintFile failed with error %s\n", debugstr_a( pcupsLastErrorString() ) );

-        pcupsFreeOptions( num_options, options );
+        printf( "filename: %s\n", unixname );

-        free( unix_doc_title );
-        free( queue );
-        free( unixname );
+        BOOL test = do_print_file(unixname);
         return ret;
     }
     else
 #endif
     {
-        return schedule_lpr( printer_name, filename );
+        printf("Schedule lpr");
+	return schedule_lpr( printer_name, filename );
     }
 }

